src/main/scala/io/github/pauljamescleary/petstore/model/PetStatus.scala

```diff
@@ -0,0 +1,21 @@
+package io.github.pauljamescleary.petstore.model
+
+sealed trait PetStatus extends Product with Serializable
+case object Available  extends PetStatus
+case object Pending    extends PetStatus
+case object Adopted    extends PetStatus
+
+object PetStatus {
+
+  def apply(name: String): PetStatus = name match {
+    case "Available" => Available
+    case "Pending"   => Pending
+    case "Adopted"   => Adopted
+  }
+
+  def nameOf(status: PetStatus): String = status match {
```

Comment:
This is more cumbersome but I prefer when my parsing return `Either` or `Option` because an invalid string could be fed in this and it would blow with a `ScalaMatchException`

Reply:
Nothing to do with your MR, though, I just saw it was like that before